pydis_site/apps/api/models/bot/scheduled_event.py

```diff
@@ -0,0 +1,78 @@
+from collections import namedtuple
+
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from .user_event import UserEvent
+
+DATETIME_RANGE = namedtuple("datetime_range", ("start", "end"))
+
+
+class ScheduledEvent(models.Model):
+    """A scheduled user event."""
+
+    user_event = models.OneToOneField(
+        UserEvent,
+        on_delete=models.CASCADE,
+        help_text="The user event which is scheduled."
+    )
+    start_time = models.DateTimeField(
+        unique=True,
+        help_text="The start time of the scheduled user event."
+    )
+    end_time = models.DateTimeField(
+        unique=True,
+        help_text="The end time of the scheduled user event."
+    )
+
+    def clean(self) -> None:
+        """
+        Check for event time overlap and if an organizer has already scheduled an event.
+
+        Note: not overriding the save() method as the save() method is not called when creating
+        model instances via the admin interface.
```

Comment:
What is point of this comment? This model doesn't have save function.

Reply:
The usual approach to perform validation is by overriding the save() method (as DRF using save() method and save() does not call clean() method). Hence, the comment.

Reply:
Have a look at this
https://github.com/python-discord/site/pull/410/files#diff-205404e5f7cd4c91e669b190befaeff1a67aec62487917fd317557e141735720R394